File: drift-static-profiler/tests/fixtures/go-gin/service/orders_service.go
package service

import (
	"errors"
	"strings"

	"example.com/orders/repo"
)

type OrdersService struct {
	orders *repo.OrdersRepository
}

func NewOrdersService(r *repo.OrdersRepository) *OrdersService {
	return &OrdersService{orders: r}
}

func (s *OrdersService) CreateOrder(email string, totalCents int64, currency string) (*repo.Order, error) {
	order := s.buildOrder(email, totalCents, currency)
	if err := s.validate(order); err != nil {
		return nil, err
	}
	if err := s.orders.Save(order); err != nil {
		return nil, err
	}
	return order, nil
}

func (s *OrdersService) buildOrder(email string, totalCents int64, currency string) *repo.Order {
	if currency == "" {
		currency = "USD"
	}
	return &repo.Order{
		CustomerEmail: email,
		TotalCents:    totalCents,
		Currency:      currency,
	}
}

func (s *OrdersService) validate(o *repo.Order) error {
	if o.TotalCents <= 0 {
		return errors.New("totalCents must be positive")
	}
	if !strings.Contains(o.CustomerEmail, "@") {
		return errors.New("invalid customerEmail")
	}
	return nil
}

File: drift-static-profiler/tests/fixtures/typescript-nestjs/src/orders/orders.service.ts
import { Injectable, BadRequestException } from '@nestjs/common';
import { CreateOrderDto } from './create-order.dto';
import { Order } from './order.entity';
import { OrdersRepository } from './orders.repository';

@Injectable()
export class OrdersService {
  constructor(private readonly repository: OrdersRepository) {}

  async createOrder(dto: CreateOrderDto): Promise<Order> {
    const order = this.buildOrder(dto);
    this.validate(order);
    return this.repository.save(order);
  }

  private buildOrder(dto: CreateOrderDto): Order {
    const order = new Order();
    order.customerEmail = dto.customerEmail;
    order.totalCents = dto.totalCents;
    order.currency = dto.currency ?? 'USD';
    return order;
  }

  private validate(order: Order): void {
    if (!order.totalCents || order.totalCents <= 0) {
      throw new BadRequestException('totalCents must be positive');
    }
    if (!order.customerEmail.includes('@')) {
      throw new BadRequestException('invalid customerEmail');
    }
  }
}
